src/org/mozilla/javascript/ScriptableObject.java

```diff
@@ -1026,44 +1026,11 @@ public static Object getDefaultValue(Scriptable object, Class<?> typeHint)
             }
 
             String methodName;
-            Object[] args;
+            Object[] args = ScriptRuntime.emptyArgs;
             if (tryToString) {
                 methodName = "toString";
-                args = ScriptRuntime.emptyArgs;
             } else {
                 methodName = "valueOf";
-                args = new Object[1];
```

Comment:
This is a lot of code and it must have been doing something -- or at least it was in some previous version of Rhino. Can you give us some context as to what's changing, what tests it fixes, and whether we think it'll break people used to older versions of Rhino?

Reply:
Yes i was surprised also about this huge amount of code used for nothing. I think this was something left over from the really beginning, but did not any real history searching.

If you like to have a test case for this you can have a look at arg-to-number.js (and all the other args-to... for dates. There is an assert that proofs the valueOf method is called without any parameters.

If you have a look at all the core classes offering valueOf methods, all these methods are defined using an arity 0 and none of them uses any argument.

Did also some search on the internet and was not able to find some spec (even old one) that has any hint about this parameter.

And finally there was no failing test after this removal.